samples/sensor/bme280/README.rst

```diff
@@ -0,0 +1,54 @@
+.. _bme280:
+
+BME280 Humidity and Pressure Sensor
+###################################
+
+Overview
+********
+
+This sample application periodically reads temperature, pressure and humidity data from
+the first available device that implements SENSOR_CHAN_AMBIENT_TEMP, SENSOR_CHAN_PRESS
+and SENSOR_CHAN_HUMIDITY. This sample checks the sensor in polling mode (without
+interrupt trigger).
+
+Building and Running
+********************
+
+This sample application uses an BME280 sensor connected to an Arduino 101 board via I2C.
+Sensor has multiple pins so you need to connect according to connection diagram given in
+`bme280 datasheet`_ at page 38.
+There are two processor cores (x86 and ARC) on the Arduino 101. You will need to
+flash both this sample's code on the ARC core (using the ``arduino101_ss`` board target),
+and stub code on the x86 core (using the ``arduino_101`` board target), as shown below.
+
+.. code-block:: console
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/sensors/bme280
+   :board: arduino_101_sss
```

Comment:
This sample doesn't build out of the box for this board. Will you please also add a dts overlay to fix this?
```
In file included from /home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:24:
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.h:111:2: error: #error "BME280 device type not specified"
 #error "BME280 device type not specified"
  ^~~~~
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c: In function 'bme280_init':
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:357:22: warning: unused variable 'data' [-Wunused-variable]
  struct bme280_data *data = dev->driver_data;
                      ^~~~
In file included from ../../../../include/sensor.h:23,
                 from /home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:11:
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c: At top level:
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:385:29: error: 'DT_BOSCH_BME280_0_LABEL' undeclared here (not in a function); did you mean 'DT_BOSCH_BMI160_0_LABEL'?
 DEVICE_AND_API_INIT(bme280, DT_BOSCH_BME280_0_LABEL, bme280_init, &bme280_data,
                             ^~~~~~~~~~~~~~~~~~~~~~~
../../../../include/device.h:107:11: note: in definition of macro 'DEVICE_AND_API_INIT'
   .name = drv_name, .init = (init_fn),     \
           ^~~~~~~~
```

Reply:
Ok, which board do your prefer? Arduino_101?

Reply:
Done. Please help me to review again. Thanks.